CopenhagenCityArchives/kbh-billeder
# Optimize existing stats api cache

**Toggl: Caching af stats API response, så top 10 siden loader hurtigere. In-memory, meget simpelt, læs max en gang hver 20. minut**
[Trello card](https://trello.com/c/Hw7bCF6y/201-caching-af-stats-api-response-s%C3%A5-top-10-siden-loader-hurtigere-in-memory-meget-simpelt-l%C3%A6s-max-en-gang-hver-20-minut)

# Changes
This MR simply optimizes the already existing cache. 
It tried to use some newer functions on the cache, but that broke the something. 
When a cached stat expires, we will try to update it, so it's ready for the users when they come calling. 

## How to test
This can be tested by setting the `kbhBillederStatsApi.cacheTTL` and set it to something like 60 (its seconds). 
Go to /top-10 on kbh-billeder. Look in the docker compose output. It should show the follwoing: 
![image](https://github.com/CopenhagenCityArchives/kbh-billeder/assets/17847632/f319cd3d-f800-422e-a8fc-68580e8a604b)
Happy testing
